internal/dataplane/kongstate/kongstate.go

```diff
@@ -193,17 +193,33 @@ func (ks *KongState) FillConsumerGroups(_ logr.Logger, s store.Storer) {
 	}
 }
 
+// servicesAsObjects returns a corev1.Service as a client.Object. It's used as a helper with lo.Map to return something
+// acceptable to functions that accept multiple client.Objects, since simply expanding the Service slice results in a
+// compiler error.
+func servicesAsObjects(svc *corev1.Service, _ int) client.Object {
+	return svc
+}
+
 func (ks *KongState) FillOverrides(
 	logger logr.Logger,
 	s store.Storer,
 	failuresCollector *failures.ResourceFailuresCollector,
 ) {
 	for i := 0; i < len(ks.Services); i++ {
 		// Services
-		ks.Services[i].override()
+		if err := ks.Services[i].override(); err != nil {
+			servicesGroup := lo.Values(ks.Services[i].K8sServices)
+			failuresCollector.PushResourceFailure(err.Error(), lo.Map(servicesGroup, servicesAsObjects)...)
+		}
 
 		// Routes
 		for j := 0; j < len(ks.Services[i].Routes); j++ {
+			// routes nested under Services here do not include their original parent object info in kongstate. translators
+			// convert this into a util.K8sObjectInfo, which includes name/namespace/GVK as strings. unfortunately we can't
+			// really convert this into a client.Object for use with the failures collector, and plumbing the original object
+			// down into the kongstate route copy looked a bit annoying. protocol validation for routes instead lives in the
+			// HTTPRoute and Ingress translators (these may override to ws/wss, whereas the others are expected to derive
+			// their protcol from the resource type alone.
 			ks.Services[i].Routes[j].override(logger)
```

Comment:
I think we're missing context why this comment is placed here. How about adding a sentence to explain:
```suggestion
      // Routes, opposed to Services, are not validated in their override method therefore we have no error check here.
			// Routes nested under Services here do not include their original parent object info in kongstate. Translators
			// convert this into a util.K8sObjectInfo, which includes name/namespace/GVK as strings. Unfortunately we can't
			// really convert this into a client.Object for use with the failures collector, and plumbing the original object
			// down into the kongstate.Route copy looked a bit annoying. Protocol validation for routes instead lives in the
			// HTTPRoute and Ingress translators (these may override to ws/wss, whereas the others are expected to derive
			// their protcol from the resource type alone).
			ks.Services[i].Routes[j].override(logger)
```